### PR TITLE
[RA-499] Created the privileges and roles

### PR DESCRIPTION
**Pull request by rahul0611**
screen shot 2016-02-05 at 9 24 16 pm

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java [hunk 1]
```diff
@@ -57,14 +59,18 @@
         public static final String APPLICATION_APPOINTMENT_SCHEDULER = "Application: Schedules Appointments";
         public static final String APPLICATION_APPOINTMENT_VIEWER = "Application: Sees Appointment Schedule";
         public static final String APPLICATION_APPOINTMENT_REQUESTER = "Application: Requests Appointments";
-        
+
         public static final String APPLICATION_RECORDS_ALLERGIES = "Application: Records Allergies";
+        public static final String APPLICATION_EDITS_EXISTING_ENCOUNTERS = "Application: Edits Existing Encounters";
+
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java]: Random unnecessary extra lines

File: api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java [hunk 2]
```diff
@@ -92,6 +98,11 @@ public void install() throws Exception {
         install(privilege(_Privilege.TASK_REFERENCEAPPLICATION_SIMPLE_VISIT_NOTE, "Able to write clinical notes"));
         install(privilege(_Privilege.TASK_REFERENCEAPPLICATION_VITALS, "Able to enter vitals"));
         install(privilege(_Privilege.TASK_MODIFY_ALLERGIES, "Able to record allergies"));
+        install(privilege(_Privilege.TASK_EMR_PATIENT_ENCOUNTER_DELETE, "Able to delete patient encounters"));
+        install(privilege(_Privilege.TASK_EMR_PATIENT_ENCOUNTER_EDIT, "Able to edit patient encounters"));
+
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java]: Random unnecessary extra lines

File: api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java [hunk 3]
```diff
@@ -144,6 +155,15 @@ public void install() throws Exception {
                 _Privilege.APP_APPOINTMENTSCHEDULINGUI_HOME,
                 "App: appointmentschedulingui.viewAppointments")));
 
+
+        install(role(_Role.APPLICATION_EDITS_EXISTING_ENCOUNTERS,
+                "Gives user the ability to edit patient encounter", idSet(), idSet(
+                        _Privilege.TASK_EMR_PATIENT_ENCOUNTER_DELETE,
+                        _Privilege.TASK_EMR_PATIENT_ENCOUNTER_EDIT)));
+
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java]: Random unnecessary extra lines

File: api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java [hunk 4]
```diff
@@ -168,6 +188,10 @@ public void install() throws Exception {
                 _Privilege.APP_APPOINTMENTSCHEDULINGUI_HOME,
                 "App: appointmentschedulingui.appointmentTypes")));
 
+
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/module/referencemetadata/RolePrivilegeMetadata.java]: Random unnecessary extra lines

**Comment by rahul0611** [comment]
Hello, 
I have committed changed.

**Comment by djazayeri** [comment]
I didn't mean to remove _all_ blank lines, just the places where your first commit had added 3-4 of them in a row...

**Comment by rahul0611** [comment]
I'm sorry, I thought I should remove all unnecessary extra lines. I will undo it.

**Pull request merged**